### DISCUSSION TITLE
Largest file size for uploads (selfhosted)?

### DISCUSSION THREAD

**Discussion by abernh** [supabase/supabase#1799] [Questions] [answered]
I'd like to implement a simple WeTransfer clone. The to be expected file sizes are up to 2-3GB.

In the "Pricing / plans" FAQ I found the following information:
> What is the largest sized file I can deploy to storage?
> A big cat like a Lion or Siberian tiger, also Ligers are fine.
1) Why? What does it mean? Why is this a funny remark? What have cats to do with file sizes?

> If you are uploading via the dashboard there is a 50MB limit.
2) Uploading via the API allows for bigger files then?

3) How about selfhosted ?
3.1) What is now the actually largest sized file I can deploy to storage?
3.2) What part of supabase (config) is setting this limitation?

**Comment by inian** [comment, marked as answer]
The largest file size you can upload is [50 MB right now](https://github.com/supabase/storage-api/blob/2e6c4ac70829ed7fa598d714813b83d29de08852/.env.sample#L10). This is the same for both the dashboard and API (just updated the docs). 

For self hosting you can change the linked parameter above to any number you want.

**Reply by inian** [reply]
We can change your upload sizes for your project in the hosted version if required. Reach out to beta@supabase.io and we can work something out.

**Reply by abernh** [reply]
That answers my question. Thanks for the fast reply and the details on where to make the adjustments. 
An adjustment in my hosted project is not necessary atm but it's good to know that one can reach out.

**Reply by lentsius-bark** [reply]
has the place for changing the limit changed? Just tried this and the error remains. Can't upload even the smallest of files (24kb)

**Reply by inian** [reply]
FILE_SIZE_LIMIT all the parameters you can change are documented [here](https://supabase.com/docs/guides/self-hosting/storage/config)

**Comment by morandalex** [comment]
Hello for self hosted, how to increase the limit? what is the "linked parameter " ?
